go/roothash/api/commitment/executor.go

```diff
@@ -54,15 +54,32 @@ func (h *ComputeResultsHeader) EncodedHash() hash.Hash {
 	return hash.NewFrom(h)
 }
 
+// ExecutorCommitmentFailure is the executor commitment failure reason.
+type ExecutorCommitmentFailure uint8
+
+const (
+	// FailureNone indicates that no failure has occurred.
+	FailureNone ExecutorCommitmentFailure = 0
+	// FailureUnknown indicates a generic failure.
+	FailureUnknown ExecutorCommitmentFailure = 1
+	// FailureStorageUnavailable indicates that batch processing failed due to
+	// storage being unavailable.
+	FailureStorageUnavailable ExecutorCommitmentFailure = 2
+)
+
 // ComputeBody holds the data signed in a compute worker commitment.
 type ComputeBody struct {
-	Header            ComputeResultsHeader   `json:"header"`
-	StorageSignatures []signature.Signature  `json:"storage_signatures"`
-	RakSig            signature.RawSignature `json:"rak_sig"`
+	Header  ComputeResultsHeader      `json:"header"`
+	Failure ExecutorCommitmentFailure `json:"failure,omitempty"`
 
 	TxnSchedSig      signature.Signature   `json:"txn_sched_sig"`
 	InputRoot        hash.Hash             `json:"input_root"`
 	InputStorageSigs []signature.Signature `json:"input_storage_sigs"`
+
+	// Optional fields (may be absent for failure indication).
+
+	StorageSignatures []signature.Signature  `json:"storage_signatures,omitempty"`
+	RakSig            signature.RawSignature `json:"rak_sig,omitempty"`
```

Comment:
So the ADR specifies that some of these fields (and in `ComputeResultsHeader`) should be made optional by making them pointers so they can be `nil`. This will require some more checks (and corresponding tests) to make sure they either are or are not nil (depending on the failure field), but it should result in smaller failure messages.

Reply:
Yeah that make sense, will update to match the ADR :+1: